packages/component-base/src/polylit-mixin.js

```diff
@@ -195,6 +195,27 @@ const PolylitMixinImplementation = (superclass) => {
       return result;
     }
 
+    constructor() {
+      super();
+      this.__hasPolylitMixin = true;
+    }
+
+    /** @protected */
+    connectedCallback() {
+      super.connectedCallback();
+
+      // Components like `vaadin-overlay` are teleported to the body element when opened.
+      // If their opened state is set as an attribute, the teleportation happens immediately
+      // after they are connected to the DOM. This means they will be outside the scope of
+      // querySelectorAll in the parent component's `firstUpdated()`. To ensure their reference
+      // is still registered in the $ map, we propagate the reference here.
+      const parentHost = this.getRootNode().host;
+      if (parentHost && parentHost.__hasPolylitMixin && this.id) {
+        parentHost.$ ||= {};
+        parentHost.$[this.id] = this;
```

Comment:
How do I reproduce the issue this resolves? I tried adding

```html
<vaadin-dialog opened></vaadin-dialog>

<vaadin-select opened></vaadin-select>
```

...on a dev page but both work fine and the `$.overlay` reference is assigned even without this change.

Reply:
It will fail when the first render is made sync: https://github.com/vaadin/web-components/pull/8290